zebra/kernel_netlink.c

```diff
@@ -454,6 +454,11 @@ static void netlink_install_filter(int sock, __u32 pid)
 		.len = array_size(filter), .filter = filter,
 	};
 
+	if (sock < 0) {
```

Comment:
I don't think this will make a difference and I'm curious in which situation the scanner found this.

If you look at the callers of this function (which is only 1) it does the proper handling to avoid passing `-1` to this function. Also I think we'd have other bigger problems if this was true (e.g. like the new assert when scheduling a `fd` set to `-1`).

Reply:
You're right. if netlink_socket() succeeds, the socket descriptor must be valid. I trusted the Coverity reasoning flow ([1]), but it is wrong unless we missed something. Please, if you close this PR, set Coverity issue 1472236 to "false positive" :-)

[1] https://scan7.coverity.com/reports.htm#v35070/p13747/fileInstanceId=58145896&defectInstanceId=12084595&mergedDefectId=1472236

Reply:
Another way to fix this is `assert(zns->netlink.sock)` right after the call to `netlink_socket`. I tend to lean towards solving even false positives in code, with the hope that one day coverity won't find anything, false positives or not.

But I agree, this is a false positive so we can solve it however.

Reply:
I like @qlyoung 's solution.

@pacovn would you mind to implement it? I think it would fit this case better.